### ISSUE TITLE
Unable to prune deleted files

### ISSUE THREAD

**Issue by CtrlAltDefeat94** [SiaFoundation/renterd#937]
### Current Behavior

I've enabled Prune Sector Roots after deleting a bunch of files. But after weeks, the sectors still aren't deleted.

Every now and then I keep getting alerts that pruning failed, and the logs are full with the same error.

The alert:
```warning: Contract pruning failed
timestamp06-02-2024 15:04:34
failed to prune contract fcid:5719ae71088e0fcfe6fd266ae88f73c5ab09d395de546e74f04c1e87263d02ca; couldn't verify contract roots proof, host ed25519:09a5972a8fe871f765a52729fab15a78a2e0a7d85a8b614de3f4295f2a87292b, version 1.6.0; host supplied invalid Merkle proof
originautopilot.autopilot
host key
ed25519:09a5...
contract ID
5719ae71088e...
hostVersion1.6.0
```

Full error from the logs:
`contract fcid:5719ae71088e0fcfe6fd266ae88f73c5ab09d395de546e74f04c1e87263d02ca, host ed25519:09a5972a8fe871f765a52729fab15a78a2e0a7d85a8b614de3f4295f2a87292b version 1.6.0, err: failed to prune contract fcid:5719ae71088e0fcfe6fd266ae88f73c5ab09d395de546e74f04c1e87263d02ca; couldn't verify contract roots proof, host ed25519:09a5972a8fe871f765a52729fab15a78a2e0a7d85a8b614de3f4295f2a87292b, version 1.6.0; host supplied invalid Merkle proof` 

### Expected Behavior

.

### Version

#1469b66

### What operating system did the problem occur on (e.g. Ubuntu 22.04, macOS 12.0, Windows 11)?

Debian Docker

### Autopilot Config

{         "contracts": {                 "set": "autopilot",                 "amount": 50,                 "allowance": "135793206078600000000000000000",                 "period": 6048,                 "renewWindow": 2016,                 "download": 196000000000,                 "upload": 994000000000,                 "storage": 10000000000000,                 "prune": true         },         "hosts": {                 "allowRedundantIPs": true,                 "maxDowntimeHours": 1440,                 "minRecentScanFailures": 10,                 "scoreOverrides": null         } }

### Bus Config

{         "default": "autopilot" } {         "hostBlockHeightLeeway": 6,         "maxContractPrice": "1000000000000000000000000",         "maxDownloadPrice": "5805101409000000000000000000",         "maxRPCPrice": "1000000000000000000000",         "maxStoragePrice": "694444444444",         "maxUploadPrice": "2902550705000000000000000000",         "migrationSurchargeMultiplier": 10,         "minAccountExpiry": 86400000000000,         "minMaxCollateral": "10000000000000000000000000",         "minMaxEphemeralAccountBalance": "1000000000000000000000000",         "minPriceTableValidity": 60000000000 } {         "minShards": 4,         "totalShards": 12 } {         "enabled": true }

### Contract Set Contracts

23

**Comment by ChrisSchinnerl** [comment]
Possibly fixed once https://github.com/SiaFoundation/hostd/pull/296 is merged and has seen some adoption

**Comment by CtrlAltDefeat94** [comment]
I'm assuming most of us on Zen have updated. No change in behaviour so far.

**Comment by peterjan** [comment]
I have been running `dev` on my node that manages 4TB and constantly uploads/downloads and deletes data. Several TB per day. That node has pruning enabled and has been running without issues for a while now. We recently merged a bunch of stuff related to pruning, all stuff I discovered from doing this test:

https://github.com/SiaFoundation/renterd/pull/1240
https://github.com/SiaFoundation/renterd/pull/1230
https://github.com/SiaFoundation/renterd/pull/1229

I think we can close this issue and re-open it if necessary.
